### ISSUE TITLE
trouble with record start and record stop

### ISSUE THREAD

**Issue by sflavier** [dev-jam/opensesame-plugin-audio_low_latency#1]
Hello.
first thanks for making this plugin.

I'm trying to use the background record tool, and i get some trouble :
1) In audio_low_latency_record_start, it's quite confusing that it ask for a file in the pool (maybe changing the filepool type in yaml to line_edit ? and some ref in code too )
I understand now that i have to type the name i want and it will try to create it (except i get an Read-only error  from line 172, the only solution is to set the full path to the file, not only from the working dir of the experiment which is perfect when running an experiment).

2) When i setup with a filename, i get the file created in the folder i ask, but when i finish the recording i get this error ( with and without the cache to ram option):

> File "/Applications/OpenSesame.app/Contents/Resources/share/opensesame_plugins/audio_low_latency_record_start/audio_low_latency_record_start.py", line 278, in record
>     wav_file.writeframes(b''.join(frames))
> TypeError: sequence item 0: expected a bytes-like object, tuple found

In the verbose information, i only get a "Initializing audio recording" and then a "sending stop signal" . may be some more verbose information are needed ?

my working environment :
system : Mac os 12.4 with apple silicon M1
opensesame : OpenSesame 3.3.12 Lentiform Loewenfeld 
audio_low_latency : version 2022.05-1

I can join the opensesesame project, and the log from the verbose option in the record init if needed.

have a nice day,

Sébastien

**Comment by dev-jam** [comment]
Hi Sébastien,

Nice to hear you are using this plugin. I was still fixing the sounddevice and pyaudio code. On linux it did work but on Windows it was really buggy. Most of the time we use pyalsaaudio on debian which works pretty stable and gives a 1 ms onset latency. I now found out I was mixing blocking mode code with callback mode code for sounddevice and pyaudio. So I first fixed that part. 

I was still looking for converting the numpy array data to byte data and your numpy line seems to work. I did not test it with a real mic yet but the wave file seems consistent now. Thanks!

I am also not really experienced yet with accepting and merging commits. Your code needed an extra if statement so it does not convert the pyaudio data. For now I am gonna merge them manually if that is ok with you and put a reference to you in the commit. And also merge this in the background record start item.

Next on the agenda is to test for stability and look into the file pool object for saving the wave files. I only used this file pool object for opening files, so it follows the default opensesame search procedure for finding the file. And I also find it curious that for playback with sounddevice I did not have to convert from byte data to numpy data, also gonna check that out.

Btw nice to see that it is also now working on the mac, never tested that before. 

Cheers!
Bob
